src/main/kotlin/build/buf/intellij/index/BufIndexes.kt

```diff
@@ -39,10 +40,14 @@ object BufIndexes {
 
     private fun <K : Any> getProjectIndexKeys(indexId: ID<K, Void>, project: Project): Collection<K> {
         val scope = GlobalSearchScope.projectScope(project)
-        return FileBasedIndex.getInstance().getAllKeys(indexId, project).filter {
+        val fileBasedIndex = FileBasedIndex.getInstance()
+        val projectFileIndex = ProjectRootManager.getInstance(project).fileIndex
+        return fileBasedIndex.getAllKeys(indexId, project).filter {
             // NOTE: getAllKeys(..., project) isn't actually filtering out keys that only exist in the project.
             // Filter the results to ensure that we only return keys that were indexed from the project's files.
-            FileBasedIndex.getInstance().getContainingFiles(indexId, it, scope).isNotEmpty()
+            // Exclude any files found in directories which are marked as excluded.
+            fileBasedIndex.getContainingFiles(indexId, it, scope)
+                .any { file -> !projectFileIndex.isExcluded(file) }
```

Comment:
Found some internal projects with a lot of test fixtures containing `buf.yaml` and `buf.lock` files and I wanted an easy way to exclude those from showing up as external libraries.

If we wanted to special case certain things we might at some point entertain ignoring indexing any files found in testdata folders in Go projects (files with a go.mod file).